src/sentry/services/hybrid_cloud/import_export/model.py

```diff
@@ -0,0 +1,228 @@
+# Please do not use
+#     from __future__ import annotations
+# in modules such as this one where hybrid cloud data models or service classes are
+# defined, because we want to reflect on type annotations and avoid forward references.
+
+from collections import defaultdict
+from enum import Enum, unique
+from typing import Dict, Literal, Optional, Set, Tuple, Union
+
+from pydantic import Field, StrictInt, StrictStr
+from typing_extensions import Annotated
+
+from sentry.backup.dependencies import (
+    ImportKind,
+    NormalizedModelName,
+    PrimaryKeyMap,
+    get_model,
+    get_model_name,
+)
+from sentry.backup.findings import Finding, InstanceID
+from sentry.backup.helpers import Filter, ImportFlags
+from sentry.backup.scopes import ExportScope, ImportScope
+from sentry.services.hybrid_cloud import RpcModel
+
+
+class RpcFilter(RpcModel):
+    """
+    Shadows `sentry.backup.helpers.Filter` for the purpose of passing it over an RPC boundary.
```

Comment:
Would it be simpler to only have a filter object that can be serialized? Having to translate between two representations of the same concepts when neither is bound to the database could help simplify the modelling in backup/restore.

Reply:
I already attempted this when originally writing this PR, but ran into some difficulty that I now wish I wrote down more clearly. I do remember time boxing that effort and then deciding that wrangling with it wasn't worth it, and leaving the TODO a few lines down on L37 to look at it later as part of the tidying up tasks for this project.